### DISCUSSION TITLE
Message on failed permissions check

### DISCUSSION THREAD

**Discussion by 0308199710520** [Rapptz/discord.py#6339] [Q&A]
Hi, im probably missing something obvious in the documentation but is there a way to send the author of a message that has failed a permissions check a direct message automatically?

I want to be able to tell them they dont have permsisions to access the command, that if they think they should be able to access the command to contact a moderator, information on the commands they can access with their permissions and that the attempt has been logged.

The content of the message isnt crazy important, just a way of sending the message would be appreciated, if there is one.

Cheers

**Comment by tk421bsod** [comment]
`on_command_error` gets passed a `Context`, so you can call `author.send(message)` on that Context to DM the person who invoked the command. `discord.ext.commands.CheckFailure` gets raised when a check fails, so check for that error, then send the message. 
([docs for CheckFailure](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#discord.ext.commands.CheckFailure), [docs for on_command_error](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#discord.on_command_error), [docs for send](https://discordpy.readthedocs.io/en/latest/api.html#discord.User.send))
